lms/djangoapps/courseware/module_render.py

```diff
@@ -971,32 +971,64 @@ def handle_xblock_callback_noauth(request, course_id, usage_id, handler, suffix=
         return _invoke_xblock_handler(request, course_id, usage_id, handler, suffix, course=course)
 
 
-class XblockCallbackView(APIView):
+@csrf_exempt
+@xframe_options_exempt
+def handle_xblock_callback(request, course_id, usage_id, handler, suffix=None):
     """
-    Class-based view for extensions. This is where AJAX calls go.
+    Generic view for extensions. This is where AJAX calls go.
 
-    Return 403 error if the user is not logged in. Raises Http404 if
-    the location and course_id do not identify a valid module, the module is
-    not accessible by the user, or the module raises NotFoundError. If the
-    module raises any other error, it will escape this function.
+    Arguments:
+        request (Request): Django request.
+        course_id (str): Course containing the block
+        usage_id (str)
+        handler (str)
+        suffix (str)
+
+    Raises:
+        HttpResponseForbidden: If the request method is not `GET` and user is not authenticated.
+        Http404: If the course is not found in the modulestore.
     """
-    authentication_classes = (SessionAuthentication, OAuth2AuthenticationAllowInactiveUser)
-    permission_classes = (IsAuthenticated,)
+    # In this case, we are using Session based authentication, so we need to check CSRF token.
+    if request.user and request.user.is_authenticated():
+        error = CsrfViewMiddleware().process_view(request, None, (), {})
+        if error:
+            return error
 
-    def get(self, request, course_id, usage_id, handler, suffix=None):
-        return _get_course_and_invoke_handler(request, course_id, usage_id, handler, suffix)
+    else:
+        authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+        authenticators = [auth() for auth in authentication_classes]
+
+        for authenticator in authenticators:
+            try:
+                user_auth_tuple = authenticator.authenticate(request)
+            except APIException:
+                log.exception(
+                    "XBlock handler %r failed to authenticate with %s", handler, authenticator.__class__.__name__
+                )
+            else:
+                if user_auth_tuple is not None:
+                    request.user, _ = user_auth_tuple
+                    break
 
-    def post(self, request, course_id, usage_id, handler, suffix=None):
-        return _get_course_and_invoke_handler(request, course_id, usage_id, handler, suffix)
+    # NOTE (CCB): Allow anonymous GET calls (e.g. for transcripts). Modifying this view is simpler than updating
+    # the XBlocks to use `handle_xblock_callback_noauth`, which is practically identical to this view.
+    if request.method != 'GET' and not (request.user and request.user.is_authenticated):
```

Comment:
@Agrendalath, in Django 1.8.x, `is_authenticated` is a method.

```suggestion
    if request.method != 'GET' and not (request.user and request.user.is_authenticated()):
```

Reply:
Ah, right, didn't notice this one.